sanic/pages/error.py

```diff
@@ -0,0 +1,104 @@
+from typing import Any, Mapping
+
+import tracerite.html  # type: ignore
+
+from html5tagger import E
+from tracerite import html_traceback, inspector  # type: ignore
+
+from sanic.request import Request
+
+from .base import BasePage
+
+
+# Avoid showing the request in the traceback variable inspectors
+inspector.blacklist_types += (Request,)
+
+ENDUSER_TEXT = """We're sorry, but it looks like something went wrong. Please try refreshing the page or navigating back to the homepage. If the issue persists, our technical team is working to resolve it as soon as possible. We apologize for the inconvenience and appreciate your patience."""  # noqa: E501
+
+
+class ErrorPage(BasePage):
+    STYLE_APPEND = tracerite.html.style
+
+    def __init__(
+        self,
+        title: str,
+        text: str,
+        request: Request,
+        exc: Exception,
+    ) -> None:
+        super().__init__()
+        # Internal server errors come with the text of the exception,
+        # which we don't want to show to the user.
+        # FIXME: Needs to be done in a better way, elsewhere
+        if "Internal Server Error" in title:
+            text = "The application encountered an unexpected error and could not continue."  # noqa: E501
+        name = request.app.name.replace("_", " ").strip()
+        if name.islower():
+            name = name.title()
```

Comment:
I don't think we should be changing the user's app name. Also, we probably should just use a generic term when `debug=False` because we do not explicitly tell our users that app names might be public. For example, I use internal names for applications all the time that are not the equivalent to what an end user might know it as.

Reply:
I would like to see that handled as public end user string, but taking that it is normally a Python identifier, it needs that pretty printing. It is shown on page header.

Reply:
It becoming public needs to be mentioned in release notes but I don't see how revealing a module name, "main", or such, even an internal codename, would be dangerous even if someone missed that happening, and can be easily fixed to follow product branding if needed.

Reply:
Okay, we can leave for now. But, I am wondering if maybe we should add a public name. Not required, but would get used in place of this if present.

```python
app = Sanic("ProjectName", public_name="Adam's Awesome API")
```

Let's tackle that outside of this context though.